middleware/auth.js

```diff
@@ -0,0 +1,23 @@
+const jwt = require('jsonwebtoken');
+const User = require('../models/user');
+
+module.exports = async (req, res, next) => {
+  try {
+    const token = req.headers.authorization.split(' ')[1]; 
+    if (!token) {
+      return res.status(401).json({ message: "Not authorized" });
+    }
+
+    const decoded = jwt.verify(token, process.env.JWT_SECRET);
+    const user = await User.findById(decoded.userId);
+
+    if (!user || token !== user.token) {
+      return res.status(401).json({ message: "Not authorized" });
+    }
+
+    req.user = user; 
+    next();
+  } catch (error) {
+    return res.status(401).json({ message: "Not authorized" });
```

Comment:
moze błąd 5xx i jakiś externall error. no chyba że token jest not valid. Bo może być też błąd na linii baza danych serwer.

Reply:
Poprawione.